documentation/theme/tutorial-theming-overlay.asciidoc

```diff
@@ -34,63 +34,54 @@ Attributes from the main component are not copied to the overlay component, e.g.
 
 The theme attribute is an exception to the above, and it is copied to the overlay component, allowing styling of individual overlays.
 
-To style an overlay component it is necessary to create a style/theme module, in the same way that is done for other components, and target the stylable parts, optionally scoped with a theme variant selector:
+To style an overlay component it is necessary to create a style CSS module, target the stylable parts, instruct flow to import it
+as a module targeting the specific component, and set the `theme` atribute to the main component in your view:
 
-.frontend/styles/my-overlay-theme.js
-[source,js]
+.frontend/styles/my-overlay-theme.css
+[source,css]
 ----
-const $_documentContainer = document.createElement('template');
-$_documentContainer.innerHTML = `<dom-module id="..." theme-for="vaadin-*-overlay">
-    <template>
-        <style>
-            :host([theme~="custom-theme-variant"]) [part~="overlay"] {
-                ...
-            }
-        </style>
-    </template>
-</dom-module>`;
-document.head.appendChild($_documentContainer.content);
+:host([theme~="custom-theme-variant"]) [part~="overlay"] {
+}
 ----
 
 .MyApplication.java
 [source,java]
 ----
 @Route(value = "")
-@JsModule("./styles/my-overlay-theme.js")
+@CssImport(value = "./styles/my-overlay-theme.css", themeFor = "vaadin-*-overlay")
 public class MyApplication extends Div {
 }
 ----
 
-In the case of vaadin-dialog-overlay
+.MyView.java
+----
+public class MyView extends VerticalLayout {
+    public MyView {
+        Dialog dialog = new Dialog();
+        dialog.getElement().setAttribute("theme", "custom-theme-variant");
```

Comment:
Didn't we recently introduce some Java API to set theme variant? `getThemeList().add("custom-theme-variant");` does the same?

Reply:
Apparently Dialog does not support it, it does not implement `HasTheme`

Reply:
This can go forward as this, but should Dialog implement HasTheme? Should we maybe use another component as an example because it don't?